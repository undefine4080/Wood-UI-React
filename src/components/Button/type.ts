import commonProps from "@base/types/commonInterface";

interface propsButton extends commonProps {
    type?: 'plain' | 'important' | 'danger' | 'success' | 'border' | 'warn' | 'line';
    onClick?: ( event: React.MouseEvent<HTMLButtonElement> ) => any;
    loading?: boolean;
}

export type { propsButton }; 