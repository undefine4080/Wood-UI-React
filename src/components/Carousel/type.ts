interface carouselEvents {

}

interface propsCarousel {
    width?: number;
    height?: number | string;
    auto?: boolean;
    orientation?: 'vertical' | 'horizon';
    time?: number;
    indicatorVisible?: boolean;
    outerIndicator?: boolean; // if the indicator is outside the carousel container
    hoverPause?: boolean;
    children?: any;
}

export type { propsCarousel };