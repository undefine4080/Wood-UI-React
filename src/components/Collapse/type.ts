import commonProps from '@common/types';
import { ReactNode } from 'react';

interface propsCollapse extends commonProps {
    expandItems?: Array<string | number>;
}

interface propsCollapseItem {
    title?: ReactNode;
    index?: number;
    size?: commonProps['size'];
    children?: ReactNode;
    onClick?: (index: number) => void;
}

export type { propsCollapse, propsCollapseItem };
