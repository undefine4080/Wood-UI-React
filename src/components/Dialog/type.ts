import { ReactElement } from 'react';

interface propsDialog {
    visible: boolean;
    title?: string;
    header?: JSX.Element | string;
    footer?: ReactElement;
    children?: any;
    mask?: boolean;
    showClose?: boolean;
    closeOnMaskClick?: boolean;
    close: () => void;
    onOpen?: () => void;
    onClose?: () => void;
}

export default propsDialog;
