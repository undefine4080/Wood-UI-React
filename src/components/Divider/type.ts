interface propsDivider {
  direction?: "horizontal" | "vertical";
  text?: string;
}

export type { propsDivider };
