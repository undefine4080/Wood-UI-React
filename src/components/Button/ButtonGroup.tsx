import { getNamedChild } from '@base/utils';
import { ReactNode } from 'react';

function ButtonGroup(props: { children: ReactNode }) {
    const buttonNodes = getNamedChild('Button', props.children);
    return <div className='wdu-button-group'>{buttonNodes}</div>;
}

export default ButtonGroup;
