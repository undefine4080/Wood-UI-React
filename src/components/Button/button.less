.wdu-button {
    box-sizing: border-box;
    padding: 0 15px;
    outline: none;
    letter-spacing: 2px;
    white-space: nowrap;
    .wdu-transition-quick();
    .wdu-radius-normal();
    .wdu-border-solid();
    .out-shadow-light();
    border-radius: 5px;

    &:hover {
        cursor: pointer;
    }

    &__small {
        .wdu-text-small();
        height: 26px;
    }

    &__normal {
        .wdu-text-normal();
        height: 36px;
    }

    &__large {
        .wdu-text-large();
        height: 46px;
    }

    &__plain {
        .wdu-bg-plain();
        color: @wdu-font-color-main;

        &:hover {
            background: linear-gradient(#f0f0f0, #e0e0e0);
        }
    }

    &__important {
        .wdu-bg-important();
        color: @wdu-color-front;

        &:hover {
            background-image: linear-gradient(#7f9bda, #506db1);
        }
    }

    &__danger {
        .wdu-bg-danger();
        color: @wdu-color-front;

        &:hover {
            background-image: linear-gradient(#e07d78, #d36661);
        }
    }

    &__success {
        .wdu-bg-success();
        color: @wdu-color-front;

        &:hover {
            background-image: linear-gradient(#bce776, #9dbe68);
        }
    }

    &__warn {
        .wdu-bg-warn();
        color: @wdu-color-front;

        &:hover {
            background-image: linear-gradient(#fff16e, #d1bf20);
        }
    }

    &__disabled {
        .wdu-disabled();
    }

    &__line {
        box-shadow: none;
        color: @wdu-font-color-main;
        background-color: @wdu-color-front;
        border: 1px solid rgb(231, 231, 231);

        &:hover {
            background-color: #f2f0f0;
        }
    }

    &__loading {
        &:hover {
            cursor: progress;
        }
    }

    &-group {
        .wdu-radius-normal();
        .wdu-border-solid();
        .out-shadow-light();
        overflow: hidden;

        .wdu-button {
            padding: 0 15px;
            outline: none;
            letter-spacing: 2px;
            white-space: nowrap;
            .wdu-transition-quick();
            border-radius: 0;
            border: none;

            &:first-child {
                border-radius: 5px 0 0 5px;
            }

            &:last-child {
                border-radius: 0 5px 5px 0;
            }

            &:focus {
                filter: brightness(0.7);
            }
        }
    }
}