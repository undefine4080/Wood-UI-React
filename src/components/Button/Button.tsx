import React from 'react';
import './button.less';
import commonOptions from '../../base/commonInterface';
interface ButtonOptions extends commonOptions
{
    type?: string;
    size?: string;
}

const PREFIX = 'wdu-button';

export function Button ( props: any )
{
    const defaultOptions: ButtonOptions = {
        type: 'plain',
        size: 'normal',
    };
    const options: ButtonOptions = Object.assign( defaultOptions, props );
    const { type, size, children } = options;
    let theClass = `${ PREFIX } ${ PREFIX }-${ type } ${ PREFIX }-${ size }`;
    if ( type === 'disabled' ) theClass += ` ${ PREFIX }-disabled`;

    return (
        <button { ...props } className={ theClass }>{ children }</button>
    );
}