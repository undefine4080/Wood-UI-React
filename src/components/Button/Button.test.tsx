import { render, fireEvent } from '@testing-library/react';
import { describe, test, expect, vi } from 'vitest';

import Button from "./Button";
import { propsButton } from './type';

describe('Button', () => {
    test('button click event', () => {
        const handleCallback = vi.fn();
        const button = render(<Button onClick={handleCallback}></Button>);
        const element = button.getByRole('button');
        fireEvent.click(element);
        expect(handleCallback).toHaveBeenCalled();
    });

    test('disable the button', () => {
        const handleClick = vi.fn();
        const button = render(<Button onClick={handleClick} disabled></Button>);
        const element = button.getByRole('button');
        fireEvent.click(element);
        expect(handleClick).not.toHaveBeenCalled();
    });

    test('the loading button', () => {
        const { container } = render(<Button loading>loading Button</Button>);
        expect(container).toMatchSnapshot();
    });
});

const defineTypes = [
    { type: 'plain' },
    { type: 'important' },
    { type: 'danger' },
    { type: 'success' },
    { type: 'border' },
    { type: 'warn' },
    { type: 'line' }
];
describe.each(defineTypes)('test the type of button', ({ type }) => {
    test(`${type} type`, () => {
        const { container } = render(<Button type={(type as propsButton['type'])}>{type} Button</Button>);
        expect(container).toMatchSnapshot();
    });
});

const defineSize = [
    { size: 'small' },
    { size: 'normal' },
    { size: 'large' },
];
describe.each(defineSize)('test the size of button', ({ size }) => {
    test(`${size} size`, () => {
        const { container } = render(<Button size={(size as propsButton['size'])}>{size} Button</Button>);
        expect(container).toMatchSnapshot();
    });
});
