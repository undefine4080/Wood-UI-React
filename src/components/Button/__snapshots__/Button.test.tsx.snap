// Vitest Snapshot v1

exports[`Button > the loading button 1`] = `
<div>
  <button
    class="wdu-button wdu-button__plain wdu-button__normal wdu-button__loading"
    type="button"
  >
    <i
      class="wdu-icon-loading"
    />
    loading Button
  </button>
</div>
`;

exports[`test the size of button > large size 1`] = `
<div>
  <button
    class="wdu-button wdu-button__plain wdu-button__large"
    type="button"
  >
    large
     Button
  </button>
</div>
`;

exports[`test the size of button > normal size 1`] = `
<div>
  <button
    class="wdu-button wdu-button__plain wdu-button__normal"
    type="button"
  >
    normal
     Button
  </button>
</div>
`;

exports[`test the size of button > small size 1`] = `
<div>
  <button
    class="wdu-button wdu-button__plain wdu-button__small"
    type="button"
  >
    small
     Button
  </button>
</div>
`;

exports[`test the type of button > border type 1`] = `
<div>
  <button
    class="wdu-button wdu-button__border wdu-button__normal"
    type="button"
  >
    border
     Button
  </button>
</div>
`;

exports[`test the type of button > danger type 1`] = `
<div>
  <button
    class="wdu-button wdu-button__danger wdu-button__normal"
    type="button"
  >
    danger
     Button
  </button>
</div>
`;

exports[`test the type of button > important type 1`] = `
<div>
  <button
    class="wdu-button wdu-button__important wdu-button__normal"
    type="button"
  >
    important
     Button
  </button>
</div>
`;

exports[`test the type of button > line type 1`] = `
<div>
  <button
    class="wdu-button wdu-button__line wdu-button__normal"
    type="button"
  >
    line
     Button
  </button>
</div>
`;

exports[`test the type of button > plain type 1`] = `
<div>
  <button
    class="wdu-button wdu-button__plain wdu-button__normal"
    type="button"
  >
    plain
     Button
  </button>
</div>
`;

exports[`test the type of button > returns undefined 1`] = `
<div>
  <button
    class="wdu-button wdu-button__plain wdu-button__normal"
    type="button"
  >
    plain
     Button
  </button>
</div>
`;

exports[`test the type of button > returns undefined 2`] = `
<div>
  <button
    class="wdu-button wdu-button__important wdu-button__normal"
    type="button"
  >
    important
     Button
  </button>
</div>
`;

exports[`test the type of button > returns undefined 3`] = `
<div>
  <button
    class="wdu-button wdu-button__danger wdu-button__normal"
    type="button"
  >
    danger
     Button
  </button>
</div>
`;

exports[`test the type of button > returns undefined 4`] = `
<div>
  <button
    class="wdu-button wdu-button__success wdu-button__normal"
    type="button"
  >
    success
     Button
  </button>
</div>
`;

exports[`test the type of button > returns undefined 5`] = `
<div>
  <button
    class="wdu-button wdu-button__border wdu-button__normal"
    type="button"
  >
    border
     Button
  </button>
</div>
`;

exports[`test the type of button > returns undefined 6`] = `
<div>
  <button
    class="wdu-button wdu-button__warn wdu-button__normal"
    type="button"
  >
    warn
     Button
  </button>
</div>
`;

exports[`test the type of button > returns undefined 7`] = `
<div>
  <button
    class="wdu-button wdu-button__line wdu-button__normal"
    type="button"
  >
    line
     Button
  </button>
</div>
`;

exports[`test the type of button > returns wdu-button__border 1`] = `
<div>
  <button
    class="wdu-button wdu-button__border wdu-button__normal"
    type="button"
  >
    border
     Button
  </button>
</div>
`;

exports[`test the type of button > returns wdu-button__danger 1`] = `
<div>
  <button
    class="wdu-button wdu-button__danger wdu-button__normal"
    type="button"
  >
    danger
     Button
  </button>
</div>
`;

exports[`test the type of button > returns wdu-button__important 1`] = `
<div>
  <button
    class="wdu-button wdu-button__important wdu-button__normal"
    type="button"
  >
    important
     Button
  </button>
</div>
`;

exports[`test the type of button > returns wdu-button__line 1`] = `
<div>
  <button
    class="wdu-button wdu-button__line wdu-button__normal"
    type="button"
  >
    line
     Button
  </button>
</div>
`;

exports[`test the type of button > returns wdu-button__plain 1`] = `
<div>
  <button
    class="wdu-button wdu-button__plain wdu-button__normal"
    type="button"
  >
    plain
     Button
  </button>
</div>
`;

exports[`test the type of button > returns wdu-button__success 1`] = `
<div>
  <button
    class="wdu-button wdu-button__success wdu-button__normal"
    type="button"
  >
    success
     Button
  </button>
</div>
`;

exports[`test the type of button > returns wdu-button__warn 1`] = `
<div>
  <button
    class="wdu-button wdu-button__warn wdu-button__normal"
    type="button"
  >
    warn
     Button
  </button>
</div>
`;

exports[`test the type of button > success type 1`] = `
<div>
  <button
    class="wdu-button wdu-button__success wdu-button__normal"
    type="button"
  >
    success
     Button
  </button>
</div>
`;

exports[`test the type of button > warn type 1`] = `
<div>
  <button
    class="wdu-button wdu-button__warn wdu-button__normal"
    type="button"
  >
    warn
     Button
  </button>
</div>
`;
